components/server/src/projects/projects-service.ts

```diff
@@ -79,6 +121,10 @@ export class ProjectsService {
     }
 
     async getBranchDetails(user: User, project: Project, branchName?: string): Promise<Project.BranchDetails[]> {
+        if (!(await this.auth.hasPermissionOnProject(user.id, "read_info", project))) {
```

Comment:
Personally, I find the `if ( await` really hard to read.

Here, it would help me visually grep this code if it looked like this:
```suggestion
		const canRead = await this.auth.hasPermissionOnProject(user.id, "read_info", project)
        if (!canRead) {
```

Reply:
Sure. Makes it more readable in this case, because the expression is a tad too long. I disagree though that we should generally not `await` within conditions or other expression context.